### DISCUSSION TITLE
Can multithreading be disabled?

### DISCUSSION THREAD

**Discussion by emrys90** [sisk-http/core#6] [Q&A] [answered]
I want to respond to requests only from the main thread. Use case being a game server, which is typically not multithreaded supported.

**Comment by CypherPotato** [comment, marked as answer]
Yes, you can set your callback to be syncronized. Static methods are synchronized to the type, and instance methods are synchronized to the instance they are contained in. Only one thread executes with the method marked Synchronized.

Example:

```cs
using Sisk.Core.Http;
using Sisk.Core.Routing;
using System.Runtime.CompilerServices;

class Program
{
    static void Main(string[] args)
    {
        HttpServer server = HttpServer.Emit(5155, 
            out HttpServerConfiguration config,
            out var host,
            out var router);

        router += new Route(RouteMethod.Get, "/", Index);

        server.Start();
        Console.WriteLine(server.ListeningPrefixes[0]);

        Thread.Sleep(-1);
    }

    [MethodImpl(MethodImplOptions.Synchronized)]
    static HttpResponse Index(HttpRequest request)
    {
        Thread.Sleep(1000);
        return new HttpResponse(200).WithContent("Hello, world!");
    }
}
```

There, `Index` will be executed in one thread at time, and other HTTP requests will join an internal queue to attend the request.

**Comment by emrys90** [comment]
Great, thank you!

**Reply by emrys90** [reply]
Will this work for NativeAOT as well though, with it being an attribute?

**Reply by CypherPotato** [reply]
Hmmm idk tho.

Need to test. But anyway, if it doesn't work, you can opt for using `lock {}` too.

**Comment by emrys90** [comment]
Is there a way to instead manually trigger when it handles requests, to fully control when it executes and which thread it runs on?

For context, I am wanting to handle requests to have it tied to the game server loop, so it runs as part of its update cycle

**Reply by CypherPotato** [reply]
It's difficult to understand what you want to achieve. Without code it becomes even more difficult.

What exactly do you want to trigger?

**Reply by emrys90** [reply]
I think I've accomplished it by just using locks, thanks though!
